sopel/modules/currency.py

```diff
@@ -1,102 +1,189 @@
+from sopel.tools import stderr
+from sopel.config.types import StaticSection, ValidatedAttribute
 
-# The Canadian central bank has better exchange rate data than the Fed, the
-# Bank of England, or the European Central Bank. Who knew?
-base_url = 'https://www.bankofcanada.ca/valet/observations/FX{}CAD/json'
+rates_fiat_json = {}
+rates_btc_json = {}
+
+
+fiat_url = 'https://api.exchangeratesapi.io/latest?base=EUR'
+fixer_url = 'http://data.fixer.io/api/latest?base=EUR&access_key={}'
+crypto_url = 'https://apiv2.bitcoinaverage.com/indices/global/ticker/short?crypto=BTC'
 regex = re.compile(r'''
-    (\d+(?:\.\d+)?)        # Decimal number
-    \s*([a-zA-Z]{3})       # 3-letter currency code
-    \s+(?:in|as|of|to)\s+  # preposition
-    ([a-zA-Z]{3})          # 3-letter currency code
-    ''', re.VERBOSE)
-
-
-def get_rate(code):
-    code = code.upper()
-    if code == 'CAD':
-        return 1, 'Canadian Dollar'
-    elif code == 'BTC':
-        btc_rate = get('https://apiv2.bitcoinaverage.com/indices/global/ticker/BTCCAD')
-        rates = btc_rate.json()
-        return 1 / rates['averages']['day'], 'Bitcoin—24hr average'
-
-    data = get(base_url.format(code))
-    name = data.json()['seriesDetail']['FX{}CAD'.format(code)]['description']
-    name = name.split(" to Canadian")[0]
-    json = data.json()['observations']
-    for element in reversed(json):
-        if 'v' in element['FX{}CAD'.format(code)]:
-            return 1 / float(element['FX{}CAD'.format(code)]['v']), name
-
-
-@commands('cur', 'currency', 'exchange')
-@example('.cur 20 EUR in USD')
-def exchange(bot, trigger):
+    ^(\d+(?:\.\d+)?)                                            # Decimal number
+    \s*([a-zA-Z]{3})                                            # 3-letter currency code
+    \s+(?:in|as|of|to)\s+                                       # preposition
+    (([a-zA-Z]{3}$)|([a-zA-Z]{3})\s)+$                          # one or more 3-letter currency code
+''', re.VERBOSE)
+
+
+def btc_rate(code, reverse=False):
+    global rates_btc_json
+
+    search = 'BTC{}'.format(code)
+
+    if search in rates_btc_json:
+        rate = rates_btc_json[search]['averages']['day']
+    else:
+        return "Sorry {} isn't currently supported".format(code)
+
+    if reverse:
+        return 1 / rate
+    else:
+        return rate
+
+
+def update_rates(bot):
+    global rates_fiat_json
+    global rates_btc_json
+
+    # If we have data that are less than 24h old, return
+    if 'date' in rates_fiat_json:
+        if time.time() - rates_fiat_json['date'] < 24 * 60 * 60:
+            return
+
+    # Update crypto rates
+    request = requests.get(crypto_url)
+    request.raise_for_status()
+    rates_btc_json = request.json()
+
+    # Update fiat rates
+    if bot.config.currency.fixer_io_key is not None:
+        request = requests.get(fixer_url.format(bot.config.currency.fixer_io_key))
+        if not request.json()['success']:
+            stderr(str(request.json()['error']))
```

Comment:
This should be a `LOGGER.error` call instead of `stderr`, which IMO should be reserved for internal bot usage only. Modules should log with `LOGGER`.

```python
from sopel.logger import get_logger
LOGGER = get_logger(__name__)
```

Reply:
Done :)